scripts/base/protocols/radius/consts.zeek

```diff
@@ -182,7 +182,7 @@ const attr_types: table[count] of string = {
 	[171] = "Delegated-IPv6-Prefix-Pool",
 	[172] = "Stateful-IPv6-Address-Pool",
 	[173] = "IPv6-6rd-Configuration"
-} &default=function(i: count): string { return fmt("unknown-%d", i); } &deprecated="Remove in v6.1.  Usage testing indicates this function is unused.";
+};
```

Comment:
I think we should reconsider those: It appears the tables aren't used in any script, but that doesn't mean they aren't useful (we're not removing them here) and having the &default here does make sense if someone was to start using the table.

In other words, a `&default=function` on a table maybe should not be considered unused?

Reply:
Yeah I thought the same thing, but took them out since they were marked to be removed. I can revert those ones.

@vpax What do you think about not marking `default` functions as unused?

Reply:
I believe that's marked as deprecated simply because there's no reference anywhere else to `attr_types`, _not_ because it erroneously gets marked as an unused function. If I'm wrong about that, then I'm wondering why the one above it works but it doesn't (and can look into that as an Issue).

Reply:
> I believe that's marked as deprecated simply because there's no reference anywhere else to attr_types, not because it erroneously gets marked as an unused function.

That's certainly possible. A quick grep shows that none of those variables are referenced anywhere else in the scripts or the source.

Reply:
I changed those functions to be marked `&is_used` instead of deprecated. I could see just removing them entirely but they're also documenting the protocol a bit so I left them in.

Reply:
> I believe that's marked as deprecated simply because there's no reference anywhere else to attr_types, not because it erroneously gets marked as an unused function.


`attr_types` not being used would be an argument for removing the whole table as Tim points out. The message about a `&default` function not being unused is pretty confusing unless you know implementation details. That's why I could see default functions being marked as implicitly used rather than explicitly needing the `&is_used` on it.

```
$ cat test.zeek 
module MyRadius;

const ids: table[count] of string = {
        [1] = "One",
} &default=function(c: count): string { return fmt("unknown-%d", c); };
$ zeek test.zeek 
warning in ./test.zeek, line 5: non-exported function does not have any callers (MyRadius::lambda_<17123251138430341678>)
```

It probably doesn't come up enough though, so `&is_used` seems fine :-)

Reply:
Thanks @awelzel for doing a direct examination. Given the confusing nature of the message, I agree that ideally it'd be suppressed.  Feel free to file an issue on it and I'll take a look in my Copious Spare Time.  Also seems fine to just go with `&is_used` for now ...